### PR TITLE
autojump-rs: new, 0.4.0

### PR DESCRIPTION
**Pull request by chenx97**
Help me make a better `beyond` script, please

### CODE REVIEW

File: extra-utils/autojump-rs/autobuild/defines [hunk 1]
```diff
@@ -0,0 +1,6 @@
+PKGNAME=autojump-rs
+PKGSEC=utils
+PKGDEP="gcc-runtime"
+BUILDDEP="cargo"
+PKGDES="A cd command that learns"
+PKGPROV=autojump
```
Comment (MingcongBai) [inline extra-utils/autojump-rs/autobuild/defines:7]: I'd add an empty line but suit yourself.

File: extra-utils/autojump-rs/autobuild/beyond [hunk 2]
```diff
@@ -0,0 +1,21 @@
+abinfo "Generating cheksum data for asset downloading ..."
+VER=22.5.3
+cat > release-$VER.sha256 << EOF
```
Comment (MingcongBai) [inline extra-utils/autojump-rs/autobuild/beyond]: Use absolute path here.

File: extra-utils/autojump-rs/autobuild/beyond [hunk 3]
```diff
@@ -0,0 +1,21 @@
...
+00daf3698e17ac3ac788d529877c03ee80c3790472a85d0ed063ac3a354c37b1 release-v$VER.tar.gz
```
Comment (MingcongBai) [inline extra-utils/autojump-rs/autobuild/beyond]: Where did this come from?

Comment (MingcongBai) [reply extra-utils/autojump-rs/autobuild/beyond]: Also absolute path here.

File: extra-utils/autojump-rs/autobuild/beyond [hunk 4]
```diff
@@ -0,0 +1,21 @@
...
+EOF
+
+abinfo "Downloading assets ..."
+wget https://github.com/wting/autojump/archive/release-v$VER.tar.gz || \
+abdie "Failed fetching release-v$VER."
+sha256sum -c release-$VER.sha256 || abdie "Failed checksuming."
```
Comment (MingcongBai) [inline extra-utils/autojump-rs/autobuild/beyond]: Absolute path here.

Comment (MingcongBai) [reply extra-utils/autojump-rs/autobuild/beyond]: You could have not generated the checksum file here. See `extra-web/google-chrome` - I find that to be cleaner.

File: extra-utils/autojump-rs/autobuild/beyond [hunk 5]
```diff
@@ -0,0 +1,21 @@
...
+
+abinfo "Extracting asset packages ..."
+tar -xf release-v$VER.tar.gz
```
Comment (MingcongBai) [inline extra-utils/autojump-rs/autobuild/beyond]: Verbose.

File: extra-utils/autojump-rs/autobuild/beyond [hunk 6]
```diff
@@ -0,0 +1,21 @@
...
+
+abinfo "Installing assets ..."
+pushd autojump-release-v$VER
```
Comment (MingcongBai) [inline extra-utils/autojump-rs/autobuild/beyond]: I generally discourage using pushd/popd. Absolute path.

File: extra-utils/autojump-rs/autobuild/beyond [hunk 7]
```diff
@@ -0,0 +1,21 @@
...
+mkdir -pv "$PKGDIR"/etc/bashrc.d
+install -Dvm644 bin/autojump.bash "$PKGDIR"/etc/bashrc.d/40-autojump-rs
+popd
+# TODO figure out where to put files necessary for autojump to work with
```
Comment (MingcongBai) [inline extra-utils/autojump-rs/autobuild/beyond]: Comments in build scripts are pointless. Use `abwarn` if attention is wanted.

File: extra-utils/autojump-rs/autobuild/beyond [hunk 8]
```diff
@@ -0,0 +1,21 @@
...
+abdie "Failed fetching release-v$VER."
+sha256sum -c release-$VER.sha256 || abdie "Failed checksuming."
+
+abinfo "Extracting asset packages ..."
+tar -xf release-v$VER.tar.gz
+
+abinfo "Installing assets ..."
+pushd autojump-release-v$VER
+mkdir -pv "$PKGDIR"/etc/bashrc.d
+install -Dvm644 bin/autojump.bash "$PKGDIR"/etc/bashrc.d/40-autojump-rs
```
Comment (MingcongBai) [inline extra-utils/autojump-rs/autobuild/beyond]: Absolute path.

**Review by MingcongBai — CHANGES_REQUESTED** [review]

**Comment by chenx97** [comment]
changes committed.

File: extra-utils/autojump-rs/autobuild/beyond [hunk 9]
```diff
@@ -0,0 +1,29 @@
...
+pushd "$PKGDIR"/usr/share/fish/completions/
+ln -sv ../../autojump/autojump.fish
+popd
+pushd "$PKGDIR"/etc/fish/conf.d
+ln -sv ../../../usr/autojump/autojump.fish
+popd
+
+abinfo "Installing assets for zsh ..."
+mkdir -pv "$PKGDIR"/usr/share/zsh/site-functions
+install -Dvm644 "$SRCDIR"/../assets/autojump-release-*/bin/_j "$PKGDIR"/usr/share/zsh/site-functions/
```
Comment (MingcongBai) [inline extra-utils/autojump-rs/autobuild/beyond:29]: That file name could be problematic. @liushuyu your insight would be much appreciated.

**Review by MingcongBai — CHANGES_REQUESTED** [review]

**Pull request merged**